command/agent/config/config_test.go

```diff
@@ -297,30 +297,30 @@ func TestLoadConfigFile_Template(t *testing.T) {
 			fixturePath: "./test-fixtures/config-template-min.hcl",
 			expectedTemplates: []*ctconfig.TemplateConfig{
 				&ctconfig.TemplateConfig{
-					Source:      strPtr("/path/on/disk/to/template.ctmpl"),
-					Destination: strPtr("/path/on/disk/where/template/will/render.txt"),
+					Source:      testhelpers.StrPtr("/path/on/disk/to/template.ctmpl"),
+					Destination: testhelpers.StrPtr("/path/on/disk/where/template/will/render.txt"),
 				},
 			},
 		},
 		"full": {
 			fixturePath: "./test-fixtures/config-template-full.hcl",
 			expectedTemplates: []*ctconfig.TemplateConfig{
 				&ctconfig.TemplateConfig{
-					Backup:         boolPtr(true),
-					Command:        strPtr("restart service foo"),
-					CommandTimeout: timeDurationPtr("60s"),
-					Contents:       strPtr("{{ keyOrDefault \"service/redis/maxconns@east-aws\" \"5\" }}"),
-					CreateDestDirs: boolPtr(true),
-					Destination:    strPtr("/path/on/disk/where/template/will/render.txt"),
-					ErrMissingKey:  boolPtr(true),
-					LeftDelim:      strPtr("<<"),
-					Perms:          fileMode(0655),
-					RightDelim:     strPtr(">>"),
-					SandboxPath:    strPtr("/path/on/disk/where"),
+					Backup:         testhelpers.BoolPtr(true),
```

Comment:
All these methods like `BoolPtr` and `StrPtr` seem like they'd be great candidates for going into an `sdk/helper/pointerutil` package (or of a similar name). AWS' SDK has a package for it because their client uses pointers for scalars a lot, because it's important to know _whether they were provided_ in addition to what they actually are. I could see these being highly reused in Vault as well.

Reply:
I agree! At this point I'm only using them in the tests, so I put them in `testhelpers`. I would be agreeable to a PR to move them to the SDK if others felt it was needed in other parts of Vault, but right now I'm only aware of them in these tests.